src/hammer-vlsi/hammer_vlsi/hammer_tool.py

```diff
@@ -1128,3 +1131,34 @@ def verbose_tcl_append(cmd: str, output_buffer: List[str]) -> None:
         """
         output_buffer.append("""puts "{0}" """.format(cmd.replace('"', '\"')))
         output_buffer.append(cmd)
+class ModeType(Enum):
+    Auto = 1
+    Empty = 2
+    Manual = 3
+    Generate = 4
+    Generated = 5
```

Comment:
As far as I'm aware, the modes we use are:
```
    Auto = 1
    Empty = 2
    Manual = 3
    Generated = 4
```

`append`, `prepend` are part of another part of Hammer and not part of this mode enum.

Also, if we are going to create this enum, I think it may be good to also use them in some of the settings.

Reply:
We should change power straps to `generated` from `generate` if we're going to use `generate`. I personally like `generate` better, but `generated` does keep all of the modes as adjectives, which also makes sense.

Reply:
Yeah, maybe we can do that as a separate issue then?

Reply:
We can just tack it on to #397 .

Reply:
Hi Edward, thanks for your comment! I checked defaults.yml and it seems that pin place mode uses `generated`, and power straps mode uses `generate`. Just to make sure, they are the same and we'll stick with `generate`, right?
Also, I'm not sure what it means to use the enum in the settings, could you explain a bit more?

Reply:
Hi Jingyi- let's use `generated` instead of `generate`. We'll change the power straps mode to `generate`.

Reply:
Yeah, so John's comment above is alluding to that problem. I think the desired action is to use `generated` and just ignore the power straps mode for now.

As for use, if you look in the code, right now it uses strings - the intention is to use this enum instead

https://github.com/ucb-bar/hammer/blob/8bcbc40d8428ad62a5f66df8817b02ce629aa048/src/hammer-vlsi/hammer_vlsi/hammer_tool.py#L896